clients/adapter/src/main/java/org/eclipse/hono/adapter/client/command/Command.java

```diff
@@ -92,6 +92,13 @@
      */
     String getGatewayId();
 
+    /**
+     * Sets the identifier of the gateway this command is to be sent to.
+     *
+     * @param gatewayId The gateway identifier.
```

Comment:
Yes, it's supported. As stated in the other comment above, since `getGatewayId` may return `null`, I think it's ok to allow resetting a command to such a state.
I've extended the javadoc here for that.